src/Microsoft.Health.Dicom.Api/Features/Filters/ApiVersionValidatorAttribute.cs

```diff
@@ -0,0 +1,36 @@
+// -------------------------------------------------------------------------------------------------
```

Comment:
The issue with using a custom filter and exception is that we break the existing versioning behavior:
- Without the flag enabled, when requesting version 2, the response doesn't include the `api-supported-versions` header as all other responses do. If you request a version that doesn't exist (e.g. 3), the `api-supported-versions` header will return `1.0-prerelease, 1, 2`.
- The new exception message doesn't match the [versioning library output](https://github.com/dotnet/aspnet-api-versioning/blob/5057b06414e5f278f33fcbd22a5cf880bf4e21fb/src/AspNet/WebApi/src/Asp.Versioning.WebApi/Dispatcher/HttpResponseExceptionFactory.cs#L45): 
  ```
    {
      "error": {
          "code": "UnsupportedApiVersion",
          "message": "The HTTP resource that matches the request URI 'https://localhost:63838/v3/studies/2.25.11973375897257723869094853169321604230' does not support the API version '3'."
      }
  }
  ```
The way around this is to remove the version 2 attribute from all controllers and then toggle adding version 2 by convention to all controllers in the [versioning setup](https://github.com/microsoft/dicom-server/blob/1e4e37175a84d23acf6f3758b295e73466d6064b/src/Microsoft.Health.Dicom.Api/Registration/DicomServerServiceCollectionExtensions.cs#L116):
```dotnet
if (dicomServerConfiguration.Features.EnableLatestApiVersion)
{
    c.Conventions.Controller<ChangeFeedController>().HasApiVersion(ApiVersion.Parse("2"));
    // manually repeat for all controllers - I haven't found a simple way to add to all controllers dynamically
}
```
This will use the existing exception response and generate headers correctly.

Reply:
Good point on the api-supported-versions return header. I looked at conventions and decided to go with filters instead, because it followed the current pattern. 
I think the best mix will be to move all the version settinging into a custom convention and control the new behind a FF